source/common/router/router.cc

```diff
@@ -549,8 +550,14 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   retry_state_ =
       createRetryState(route_entry_->retryPolicy(), headers, *cluster_, config_.runtime_,
                        config_.random_, callbacks_->dispatcher(), route_entry_->priority());
-  do_shadowing_ = FilterUtility::shouldShadow(route_entry_->shadowPolicy(), config_.runtime_,
-                                              callbacks_->streamId());
+
+  // Determine which shadow policies to use. It's possible that we don't do any shadowing due to
+  // runtime keys.
+  for (const auto& shadow_policy : route_entry_->shadowPolicies()) {
+    if (FilterUtility::shouldShadow(shadow_policy, config_.runtime_, callbacks_->streamId())) {
+      active_shadow_policies_.push_back(std::cref(*shadow_policy.get()));
```

Comment:
I don't think you need both `*` and `.get()`: calling `*` on the unique_ptr returns a ref to the value, no need to first get the raw  pointer. So `*shadow_policy` should be sufficient iiuc.

Reply:
doh, fixed